src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const people = [...document.querySelectorAll('li')];
+
+function sortBySalary(arr) {
+  return arr.sort(
+    (person1, person2) =>
+      +person2.getAttribute('data-salary').replace(/[$,]/g, '') -
+      +person1.getAttribute('data-salary').replace(/[$,]/g, ''),
+  );
+}
+
+function makeObj(arr) {
+  const arrPeople = [];
+
+  for (const person of arr) {
+    const objPerson = {};
+
+    objPerson.name = person.textContent.trim();
+    objPerson.position = person.getAttribute('data-position');
+    objPerson.salary = person.getAttribute('data-salary');
+    objPerson.age = person.getAttribute('data-age');
+
+    arrPeople.push(objPerson);
+  }
+
+  return arrPeople;
+}
+
+sortBySalary(people);
+makeObj(people);
```

Comment:
The `sortBySalary` function sorts the `people` array, but the sorted array is not used in the `makeObj` function. To ensure the objects are created from the sorted list, you should pass the result of `sortBySalary(people)` to `makeObj`.